Mathlib/Data/Matrix/Basic.lean

```diff
@@ -524,6 +524,9 @@ theorem mapMatrix_comp (f : β →+* γ) (g : α →+* β) :
     f.mapMatrix.comp g.mapMatrix = ((f.comp g).mapMatrix : Matrix m m α →+* _) :=
   rfl
 
+theorem mapMatrix_diagonal (f : α →+* β) (d : m → α) :
+    f.mapMatrix (diagonal d) = diagonal (f ∘ d) := by simp
```

Comment:
I'm of the opinion we shouldn't have this lemma, since it's inconsistent not to have it for all the other homs, and it only chains together two existing lemmas.

Reply:
I took the liberty of deleting it.